tests/unit/test_config_provider.py

```diff
@@ -667,11 +723,102 @@ def _create_os_environ_patcher(self):
 
     def test_config_store_deepcopy(self):
         config_store = ConfigValueStore()
-        config_store.set_config_provider('foo', ConstantProvider('bar'))
+        config_store.set_config_provider(
+            'constant_value', ConstantProvider('ABC')
+        )
         config_store_copy = copy.deepcopy(config_store)
-        config_store_copy.set_config_provider('fizz', ConstantProvider('buzz'))
-        assert config_store.get_config_variable('fizz') is None
-        assert config_store_copy.get_config_variable('foo') == 'bar'
+        config_store_copy.set_config_provider(
+            'constant_value_copy', ConstantProvider('123')
+        )
+        assert config_store.get_config_variable('constant_value_copy') is None
+        assert config_store_copy.get_config_variable('constant_value') == 'ABC'
+
+    # @pytest.fixture
+    # def config_value_store(self):
+    #     return self._create_config_value_store()
+
+    @pytest.mark.parametrize(
+        'defaults_mode',
+        ['standard', 'in-region', 'cross-region', 'mobile', 'auto'],
```

Comment:
I think I may have not been clear here by my previous comment. I meant parameterize by config store parameter name (e.g. `retry_mode`, `s3`, `sts_regional_endpoints` ). There is a bunch of duplicate code here around retrieving config providers and making assertions and parameterization will allow us to make the test body much simpler.